implementations/rust/ockam/ockam_app/src/linux_url_plugin.rs

```diff
@@ -0,0 +1,63 @@
+use log::{info, warn};
+use std::io::Read;
+use std::os::unix::fs::PermissionsExt;
+use std::os::unix::net::UnixListener;
+use std::time::Duration;
+
+use tauri::{
+    async_runtime::spawn,
+    plugin::{Builder, TauriPlugin},
+    Manager, Runtime,
+};
+use tokio::time::sleep;
+
+pub(crate) const OCKAM_OPEN_URL_SOCK: &str = "/tmp/.ockam-open-url-sock";
+const ONLY_WRITE_FROM_USER_PERMISSIONS: u32 = 0o200;
+
+pub(crate) fn init<R: Runtime>() -> TauriPlugin<R> {
+    Builder::new("linux-url")
+        .setup(|app, _api| {
+            //bind fails if the file already exists
+            let _ = std::fs::remove_file(OCKAM_OPEN_URL_SOCK);
+            let listener = UnixListener::bind(OCKAM_OPEN_URL_SOCK)
+                .unwrap_or_else(|_| panic!("cannot listener on {OCKAM_OPEN_URL_SOCK}"));
+            //only allow the current user to write to the socket
+            std::fs::set_permissions(
+                OCKAM_OPEN_URL_SOCK,
+                std::fs::Permissions::from_mode(ONLY_WRITE_FROM_USER_PERMISSIONS),
+            )
+            .unwrap_or_else(|_| panic!("cannot set permissions on {OCKAM_OPEN_URL_SOCK}"));
+
+            let handle = app.clone();
+            spawn(async move {
+                //wait a bit to let the app start
+                sleep(Duration::from_millis(250)).await;
+
+                //check if we had an ockam:// argument passed to us
+                if let Some(url) = std::env::args().nth(1) {
+                    if url.starts_with("ockam:") {
+                        info!("received url: {}", url);
+                        handle.trigger_global(crate::app::events::URL_OPEN, Some(url));
+                    } else {
+                        warn!("ignored argument: {}", url);
+                    }
+                }
+
+                for stream in listener.incoming().flatten() {
```

Comment:
Should this do any of the following, either eventually or during this pass?
- Spawn a tokio task to handle these asynchronously using `tauri::async_runtime`
- Reuse the buffer for the lifetime of a single connection
- Allow the connection to send multiple sequential commands
- Defend against input that is some valid URL followed by some arbitrary content that may be hostile? (I'm imagining something slightly analogous to SQL injection, if we had some vulnerable code within the event handler(s) that trigger by `...URL_OPEN`.)

If the multiple-commands is an intentional no, then can we deliberately close out the client socket after the `trigger_global`?

Reply:
> Spawn a tokio task to handle these asynchronously using tauri::async_runtime

Absolutely, this is a blunder on my side! fixing it right away

> Reuse the buffer for the lifetime of a single connection
> Allow the connection to send multiple sequential commands

A connection should be used only once (when the second process starts) to pass the URL and that's it. I don't think there is another reason to use the connection for now.

> If the multiple-commands is an intentional no, then can we deliberately close out the client socket after the trigger_global?

Yes, I can manually call `close` rather than waiting for the object to be dropped to better convey the intention.

Reply:
ops, `close` doesn't exist, I've added an explicit drop